src/v0/auditLogs.ts

```diff
@@ -0,0 +1,131 @@
+import qs from 'qs'
+import { Client } from '../client'
+import * as errors from '../errors'
+
+export interface AuditsOptions {
+  user?: string
+  base?: string
+}
+
+export interface AuditsQuery {
+  type?: string | string[]
+  limit?: number
+  cursor_field?: string
+  embed?: string | string[]
+  uri?: string
+  branch?: string
+  register?: string
+  staff?: string
+  start?: string
+  end?: string
+}
+
+export interface AuditsMetaQuery {
+  type?: string | string[]
+}
+
+export interface AuditLogsGetOneRequestObject {
+  auditLogId: string
+  query?: AuditsQuery
+}
+
+export interface AuditsResponse {
+  data?: object[]
+  metadata?: object
+  msg?: string
+  next?: () => Promise<AuditsResponse>
+}
+
+export class AuditLogs {
+  endpoint: string
+  http: Client
+  public options: AuditsOptions
+
+  constructor(options: AuditsOptions, http: Client) {
+    this.options = options
+    this.http = http
+
+    this.endpoint = '/api/v0/audits'
+    this.options.base = this.options.base || 'https://api.tillhub.com'
+  }
+
+  getAll(q?: AuditsQuery | undefined): Promise<AuditsResponse> {
+    return new Promise(async (resolve, reject) => {
+      let next
+      let uri
+
+      try {
+        if (q && q.uri) {
+          uri = q.uri
+        } else {
+          uri = `${this.options.base}${this.endpoint}/${this.options.user}/logs`
+        }
+
+        const queryString = qs.stringify(q)
+        if (queryString) {
+          uri = `${uri}?${queryString}`
```

Comment:
As we talked F2F, please use the `UriHelper` class to construct the uri.

Reply:
I refactored it